### PR TITLE
Add Token model and view for managing API Tokens

### LINKED ISSUE #126: Manage API Token
**Is your feature request related to a problem? Please describe.**

Right now API Tokens are stored in the [User model](https://github.com/jointwt/twtxt/blob/master/internal/models.go#L54) just simple as a list of strings. This isn't great for a number of reasons as we can't easily store any additional data on the token without decrypting the entire JWT such as the expiry date/time or the device / user-agent, and this list is currently unbounded.

**Describe the solution you'd like**

We need to define a new model for storing and managing API Tokens, let's called it `Token` which at a minimum stores:

- `Value` -- The actual JWT token value which stored encrypted/signed data as given to the client using it.
- `UserAgent` -- The name of the client and version number that initially created the token.
- `CreatedAt` -- The date/time the token was created
- `ExpiresAt` -- The date/time the token expires (_if ever_)

In the [/settings](https://twtxt.net/settings) view for user accounts on [Twt.social](https://twt.social) pods we want to be able to:

- List the currently in-use API Tokens associated with the user's account
- Allow the user to delete tokens

Additionally there should be a background job, let's call it `DeleteOldTokensJob` that:

- Iterates over all stored tokens
- Checks if any have expired (_if any_)
- Deletes any expired tokens

**Describe alternatives you've considered**

N/A - The current implementation sucks.

**Additional context**

N/A -- Open to other ideas for token management, but this is the basics we need.

### PR DESCRIPTION
**Pull request by danielcooperxyz**
##### Summary
Provide functionality for managing user api tokens - Closes #126 

##### Component Name
API/User settings

### CODE REVIEW

File: internal/api.go [hunk 1]
```diff
@@ -71,16 +71,24 @@ func (a *API) initRoutes() {
...
 		log.WithError(err).Error("error creating signed token")
-		return "", err
+		return nil, err
 	}
-	return tokenString, nil
+
+	tkn := &Token{
+		Value:     tokenString,
+		UserAgent: r.UserAgent(),
+		CreatedAt: createdAt,
```
Comment (prologic) [inline internal/api.go:88]: Might want to store the a `ExpiresAt` too so we can write a `DeleteOldTokens` in `internal/jobs.go`? Mainly so we don't have to worry about decrypting the `.Value` (`.jwt.Token`) and doing a whole bunch of interface assertions just to figure out when it expires.

File: internal/models.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ type User struct {
 	IsFollowingPubliclyVisible bool   `default:"true"`
 
 	Feeds  []string `default:"[]"`
-	Tokens []string `default:"[]"`
+	Tokens []*Token `default:"[]"`
```
Comment (prologic) [inline internal/models.go:54]: Hmm can we use a foreign `Token` model here and key into something like `/tokens/<sha>` where we store the actual `Token`. `User.Tokens` can then just be a list of "sha" / ids of tokens the user account still has associated with their account?

File: internal/templates/settings.html [hunk 3]
```diff
@@ -95,6 +95,36 @@ <h2>
...
+          <tbody>
+            {{range $val := .User.Tokens}}
+            <tr>
+              <td>{{$val.UserAgent}}</td>
+              <td>{{$val.CreatedAt}}</td>
+              <td>{{$val.ExpiresAt}}</td>
+              <td>
+                <form action="/delete" method="POST" onsubmit="return confirm('Are you sure you want to delete this token? This cannot be undone!');">
+                  <button type="submit" data-tooltip="Delete" class="outline secondary">
+                    <i class="icss-x"></i>
```
Comment (prologic) [inline internal/templates/settings.html:118]: Gmmm 🤔 I _think_ you're using a new icon here we don't have in the `./internal/static/css/icons.css` bundle (_we don't ship the whole thing!_) Can you add it?

**Review by prologic — COMMENTED** [review]
This is a good start! 😎

**Comment by prologic** [comment]
@danielcooperxyz friendly ping 👍

File: internal/api.go (hunk 1 above)
Comment (danielcooperxyz) [reply internal/api.go:88]: I wasn't sure where to take the Expiry time from when I wrote this.

File: internal/models.go (hunk 2 above)
Comment (danielcooperxyz) [reply internal/models.go:54]: That could remain as a list of keys. Would be nice if the datastore would handle that under the hood. It seems like this mechanism stores and retrieves from the database as it should currently. What hash are you referring to in terms of an id? Is it the jwt value, or a SHA256 hash of the token model?

File: internal/templates/settings.html (hunk 3 above)
Comment (danielcooperxyz) [reply internal/templates/settings.html:118]: This is the same icon as the subscript icon in the twt partial, so it shows up when I test the application at the moment.

File: internal/api.go (hunk 1 above)
Comment (prologic) [reply internal/api.go:88]: Let's worry about this later, given #139 we have to rethink the way API tokens work anyway. But we still need a way to "manage" them.

File: internal/models.go (hunk 2 above)
Comment (prologic) [reply internal/models.go:54]: Can we use the [Token.Signature](https://pkg.go.dev/github.com/dgrijalva/jwt-go?tab=doc#Token) as the key here?

File: internal/templates/settings.html (hunk 3 above)
Comment (prologic) [reply internal/templates/settings.html:118]: OH so you're right! D'oh :) Sorry!

**Review by prologic — APPROVED** [review]
I _think_ we can merge this as soon as you've fixed the Token keys and amended how we store those in the user model.

**Comment by danielcooperxyz** [comment]
I think I've finished all those changes on the web functionality, although the view might benefit from being tidier. The only functionality left to create is the background task unless it's better to put it in a different PR to keep reviews simple.
API Token View

File: internal/bitcask_store.go [hunk 4]
```diff
@@ -251,3 +251,41 @@ func (bs *BitcaskStore) GetAllSessions() ([]*session.Session, error) {
 
 	return sessions, nil
 }
+
+func (bs *BitcaskStore) GetUserTokens(user *User) ([]*Token, error) {
+	tokens := []*Token{}
+	for _, signature := range user.Tokens {
+
+		fmt.Printf("looking up %s\n", signature)
```
Comment (prologic) [inline internal/bitcask_store.go:259]: Please drop any debug `fmt.Print()` calls you may have left around :)

**Review by prologic — APPROVED** [review]
LGTM 👍

File: internal/bitcask_store.go (hunk 4 above)
Comment (prologic) [inline internal/bitcask_store.go:259]:
```suggestion
```

**Pull request merged**